### PR TITLE
enable Node authorizer and NodeRestriction admission controller

### PR DESCRIPTION
**Pull request by mikedanese**
Fixes https://github.com/kubernetes/kubernetes/issues/46999
Fixes https://github.com/kubernetes/kubernetes/issues/47135

```release-note
gce kube-up: The `Node` authorization mode and `NodeRestriction` admission controller are now enabled
```

### CODE REVIEW

**Comment by luxas** [comment]
@mikedanese BTW, have we enabled the Node Authorizer in kubeadm?

File: pkg/auth/nodeidentifier/default.go [hunk 1]
```diff
@@ -55,10 +55,11 @@ func (defaultNodeIdentifier) NodeIdentity(u user.Info) (string, bool) {
 	}
 
 	userName := u.GetName()
-	nodeName := ""
-	if strings.HasPrefix(userName, nodeUserNamePrefix) {
-		nodeName = strings.TrimPrefix(userName, nodeUserNamePrefix)
+	if !strings.HasPrefix(userName, nodeUserNamePrefix) {
```
Comment (liggitt) [inline pkg/auth/nodeidentifier/default.go]: update godoc to match and move the username check before the loop over the groups

File: cluster/gce/gci/configure-helper.sh [hunk 2]
```diff
@@ -1123,7 +1123,7 @@ function start-kube-apiserver {
   fi
 
 
-  local authorization_mode="RBAC"
+  local authorization_mode="RBAC,Node"
```
Comment (liggitt) [inline cluster/gce/gci/configure-helper.sh]: I would probably put the Node authorizer first

File: cluster/gce/config-default.sh [hunk 3]
```diff
@@ -217,7 +217,7 @@ fi
 
 # Admission Controllers to invoke prior to persisting objects in cluster
 # If we included ResourceQuota, we should keep it at the end of the list to prevent incrementing quota usage prematurely.
-ADMISSION_CONTROL=NamespaceLifecycle,LimitRanger,ServiceAccount,PersistentVolumeLabel,DefaultStorageClass,DefaultTolerationSeconds,ResourceQuota
+ADMISSION_CONTROL=NamespaceLifecycle,LimitRanger,ServiceAccount,PersistentVolumeLabel,DefaultStorageClass,DefaultTolerationSeconds,NodeRestriction,ResourceQuota
```
Comment (liggitt) [inline cluster/gce/config-default.sh]: does this apply to all the gce clusters? what about ubuntu, container-linux, etc?

Comment (liggitt) [inline cluster/gce/config-default.sh]: I guess with the change to treat things without a `system:node:...` formatted name as non-nodes, this is safe

File: pkg/auth/nodeidentifier/default.go (hunk 1 above)
Comment (liggitt) [inline pkg/auth/nodeidentifier/default.go]: also, this'll require updating unit tests

File: plugin/pkg/auth/authorizer/rbac/bootstrappolicy/controller_policy.go [hunk 4]
```diff
@@ -277,6 +277,7 @@ func init() {
 		Rules: []rbac.PolicyRule{
 			rbac.NewRule("get", "list", "watch").Groups(certificatesGroup).Resources("certificatesigningrequests").RuleOrDie(),
 			rbac.NewRule("update").Groups(certificatesGroup).Resources("certificatesigningrequests/status", "certificatesigningrequests/approval").RuleOrDie(),
+			rbac.NewRule("create").Groups(authorizationGroup).Resources("subjectaccessreviews").RuleOrDie(),
```
Comment (liggitt) [inline plugin/pkg/auth/authorizer/rbac/bootstrappolicy/controller_policy.go]: needs bootstrap yaml updated

**Comment by mikedanese** [comment]
Fixed. PTAL

**Comment by mikedanese** [comment]
@liggitt good question, I'm going to do kubeadm in a followup so this gets in a starts getting tested.

**Comment by liggitt** [comment]
Sounds good

**Comment by mikedanese** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by liggitt** [comment]
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by fejta** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this
ref: #46827

**Comment by liggitt** [comment]
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Comment by luxas** [comment]
@mikedanese Although the release note says this has kubeadm changes I don't see any...

**Comment by liggitt** [comment]
kubeadm changes were split out on questions around version support

**Comment by luxas** [comment]
@liggitt in a conversation where?

**Comment by liggitt** [comment]
> @liggitt in a conversation where?

On the commit that added it, then got removed from this PR. Isn't github wonderful? :)

The question was around what versions of kube kubeadm 1.7 will install, which can affect which authorization mode and admission plugins it is able to assume exist.

**Comment by luxas** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Comment by liggitt** [comment]
Hmm, see
```
I0604 12:38:57.299041       5 node_authorizer.go:112] NODE DENY: node e2e-34208-minion-group-qtg9 cannot get unknown secret e2e-tests-replicaset-d448v/default-token-t80k0
I0604 12:38:57.299164       5 rbac.go:114] RBAC DENY: user "system:node:e2e-34208-minion-group-qtg9" groups ["system:nodes" "system:authenticated"] cannot "get" resource "secrets" named "default-token-t80k0" in namespace "e2e-tests-replicaset-d448v"
```

Will look in the morning

**Comment by mikedanese** [comment]
@k8s-bot test this

**Comment by mikedanese** [comment]
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Comment by luxas** [comment]
A bunch of conformance tests for volumes flaked for the nth time. cc @kubernetes/sig-storage-bugs 

@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Comment by liggitt** [comment]
looks like something significant merged around 7pm EST on 5/27 that caused configmap and projected volume e2e tests to start failing:

https://storage.googleapis.com/k8s-gubernator/triage/index.html?text=configmap
https://storage.googleapis.com/k8s-gubernator/triage/index.html?text=projected
https://storage.googleapis.com/k8s-gubernator/triage/index.html?text=secret

**Comment by liggitt** [comment]
Added some tracing on top of this in https://github.com/kubernetes/kubernetes/pull/46941

Seeing a weird one second delay for the in-apiserver pod watch to observe a newly bound pod. The kubelet observes it instantly and tries to fetch/mount the secret ~1 second before the local watch is told the pod is bound to the node. Continuing to investigate.

**Comment by dchen1107** [comment]
The release team: Please update the pr description by providing the associated issue for this pr for the final approval.

**Comment by liggitt** [comment]
projected volume test failures are because of https://github.com/kubernetes/kubernetes/issues/47020

**Comment by liggitt** [comment]
The bug in the projected volume plugin will turn the queue red if this merges... holding until the fix for #47020 gets reviewed/approved in the morning

**Comment by liggitt** [comment]
projected volume fix merged, unblocking this

**Comment by liggitt** [comment]
@k8s-bot test this

**Comment by liggitt** [comment]
@k8s-bot pull-kubernetes-kubemark-e2e-gce test this

**Comment by liggitt** [comment]
unit test flaked on https://github.com/kubernetes/kubernetes/issues/42289
@k8s-bot pull-kubernetes-unit test this

**Comment by liggitt** [comment]
ns deletion is flaking badly across PRs. tracking in https://github.com/kubernetes/kubernetes/issues/47135

**Comment by liggitt** [comment]
I give up. I think this test is just red until #47135 is resolved.

**Comment by liggitt** [comment]
fix for https://github.com/kubernetes/kubernetes/issues/47135 merged, let's see how we do now

**Comment by liggitt** [comment]
https://github.com/kubernetes/kubernetes/pull/47199 might have helped incrementally

**Comment by luxas** [comment]
`Quota 'SUBNETWORKS' exceeded.  Limit: 150.0` was the failure on gce-etcd3...

**Comment by liggitt** [comment]
fix the title on the namespace worker commit.

cc @deads2k @ncdc @wojtek-t on the namespace deletion worker change. I'm in favor of it... experimentally, it cut cleanup time in half on the e2e, *and* we saw peak times to sync individual namespaces cut in half as well, indicating the controllers had QPS and CPU headroom and were blocking on something else.

controller sync time before:
screen shot 2017-06-12 at 9 34 20 pm

controller sync time after:
screen shot 2017-06-12 at 9 34 38 pm

**Comment by derekwaynecarr** [comment]
+1 the worker bump to 10

File: cluster/gce/gci/configure-helper.sh [hunk 5]
```diff
@@ -1288,7 +1288,7 @@ function start-kube-apiserver {
   fi
 
 
-  local authorization_mode="RBAC"
+  local authorization_mode="Node,RBAC"
```
Comment (liggitt) [inline cluster/gce/gci/configure-helper.sh]: I haven't kept up with the cluster/gce/... deployments...

this line also appears in gce/container-linux, gce/cos, gce/ubuntu as well... do any of those need to be updated as well?

Comment (mikedanese) [inline cluster/gce/gci/configure-helper.sh]: gce/cos is a symlink to gce/gci. I'm not sure if gce/container-linux is tested anywhere. A quick search of testgrid didn't turn anything up. I'm not inclined to change anywhere that is not being tested.

Comment (liggitt) [inline cluster/gce/gci/configure-helper.sh]: oh, ok

Comment (mikedanese) [inline cluster/gce/gci/configure-helper.sh]: @roberthbailey who maintains container-linux?

Comment (roberthbailey) [inline cluster/gce/gci/configure-helper.sh]: According to the [OWNERS file](https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/container-linux/OWNERS), it's @euank, @yifan-gu, and @ethernetdan.

File: cmd/kube-controller-manager/app/options/options.go [hunk 6]
```diff
@@ -67,7 +67,7 @@ func NewCMServer() *CMServer {
 			ConcurrentJobSyncs:                              5,
 			ConcurrentResourceQuotaSyncs:                    5,
 			ConcurrentDeploymentSyncs:                       5,
-			ConcurrentNamespaceSyncs:                        5,
+			ConcurrentNamespaceSyncs:                        10,
```
Comment (roberthbailey) [inline cmd/kube-controller-manager/app/options/options.go]: Is 10 enough?

**Comment by mikedanese** [comment]
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Pull request merged**